crypto/rand/drbg_lib.c

```diff
@@ -78,29 +78,29 @@ typedef struct drbg_nonce_global_st {
 static const char ossl_pers_string[] = DRBG_DEFAULT_PERS_STRING;
 
 #define RAND_DRBG_TYPE_FLAGS    ( \
-    RAND_DRBG_FLAG_MASTER | RAND_DRBG_FLAG_PUBLIC | RAND_DRBG_FLAG_PRIVATE )
+    RAND_DRBG_FLAG_PARENT | RAND_DRBG_FLAG_PUBLIC | RAND_DRBG_FLAG_PRIVATE )
 
-#define RAND_DRBG_TYPE_MASTER                     0
+#define RAND_DRBG_TYPE_PARENT                     0
 #define RAND_DRBG_TYPE_PUBLIC                     1
 #define RAND_DRBG_TYPE_PRIVATE                    2
 
 /* Defaults */
 static int rand_drbg_type[3] = {
-    RAND_DRBG_TYPE, /* Master */
+    RAND_DRBG_TYPE, /* Parent */
     RAND_DRBG_TYPE, /* Public */
     RAND_DRBG_TYPE  /* Private */
 };
 static unsigned int rand_drbg_flags[3] = {
-    RAND_DRBG_FLAGS | RAND_DRBG_FLAG_MASTER, /* Master */
+    RAND_DRBG_FLAGS | RAND_DRBG_FLAG_PARENT, /* Main */
```

Comment:
Main is a better idea, would you mind using "main" instead of "parent"?  My thinking is that "main" would indicate that it's the top or root DRBG, while a "parent" can be somewhere in the middle of the chain, so not quite as definitive if you see what I mean...

Reply:
Wouldn't "root" be even better?

Reply:
Either root or parent is fine to replace master; you guys decide.

Reply:
I'm actually a little dubious that "master" on its own is offensive, whereas it is considered as such when put together with "slave" (for obvious reasons)...

However, I had a quick look at https://en.wikipedia.org/wiki/Master/slave_(technology), and other proposed substitutes seem to be "main" and "primary", primarily.  I see that as an argument for choosing "main" in our case.

Reply:
Main implies it is somewhat the one that is mainly used. Root or primary please.

Reply:
I'm kind of with @levitte....is "master" actually offensive when using it in the context of "master, public, private"? By avoiding this particular rename we don't have to make any changes to public API symbols - which would be preferable.

Reply:
...or, if so, would we also propose to rename our "master" branch in git??

Reply:
This could end up being an interesting discussion in and of itself...  just wanted to point out that "master" as a term most likely lives in the fuzzy space between the categorical "offensive" / "not offensive" classifications, and I really don't have enough of a clue to know which way to lean on this one.  That's not a very strong argument for either way to move from here.

I'll also note that, while this does touch public API symbols, almost all of them are new in 3.0.  Also, if they are deprecated in the RAND for providers PR, this change becomes moot.

Reply:
It's easy to do the good thing, even if some feel that "it's not necessary."

Of course, I'm not doing anything here until the hold is removed. It's not a good look, folks.

Reply:
BTW, to address @mattcaswell's concern: renaming something via a define in a public header file (as @levitte proposed) meets the "just recompile" bar that the project has publically set.

Reply:
Although there is an instance where master has a negative connotation, I don't think the word should be replaced - as it is a general word describing 'having talent' in pretty much any area. (e.g Headmaster, master-chef).

I have no objection to replacing words that associate a color with good or bad.

Reply:
Languages evolve, as do people's sensitivities. The dominant definition of master, when used as a known, is someone who has others working for them, such as slaves. The dominant definition when used as an adjective is as you said, having talent.  But the main point of the currently-labelled "master DRBG" is *not* that it has talent that distinguishes it -- it's just an instance of the exact same code. Does the different entropy source make it more talented that a  private or public DRBG?

So really, the "polite" definition isn't accurate, and the "bad" definition shouldn't be used. But even if it's just unfair collatoral damage, why is it important to hold on to a term that some feel, even if you disagree or think they are wrong, exclusionary?  What is the harm in this minor change to make a word actually more accurate? Why is it more important to be "right" than to be "good"?

Reply:
It this context it means it has authority over others (which it what it means here).
The point is the word is in the public API, so aliasing it really solves nothing.

Reply:
Does that mean that getrandom has authority over the main DRBG?  Or is getrandom just an input? And likewise, then, child DRBG's have another one, the parent, that provides input. What authority does the main DRBG have?

The word is in the public API, yes.  "Most programs will just require recompiling" is the public compatibility statement, and this PR will not add to that burden.

Reply:
As mentioned in a separate comment, "master/minion" is also completely fine terminology that preserves the intention that "this thing is controlling the other thing".

Reply:
_Master_ is in the public APIs already.  We'd have to change it via the deprecation policy.

I'm open to others names: _primal_, _initial_, _basal_, _chief_, _primary_, _dominus_, _foremost_, _principle_, _prime_, _doyen_, _best_, _first_, _suzerain_, _seed_, ...

The entire RAND framework needs some polishing @mspncp and I have some convergent ideas as to the direction we think it should go.  Nonetheless, `RAND_bytes()` and `RAND_priv_bytes()` must remain and be the primary calls -- I suspect the rest is rarely used.

Reply:
As for the deprecation policy, I believe these lines are enough:
```
diff --git a/include/openssl/rand_drbg.h b/include/openssl/rand_drbg.h
index 5ee93a3eab..382d472414 100644
--- a/include/openssl/rand_drbg.h
+++ b/include/openssl/rand_drbg.h
@@ -112,6 +112,10 @@ int RAND_DRBG_set_reseed_defaults(
                                   time_t child_reseed_time_interval
                                   );
 
+# ifndef OPENSSL_NO_DEPRECATED_3_0
+#  define OPENSSL_CTX_get0_master_drbg(ctx) OPENSSL_CTX_get0_parent_drbg(ctx)
+#  define RAND_DRBG_get0_master() RAND_DRBG_get0_parent()
+# endif
```

If you want a HISTORY entry in ```doc/man3/RAND_DRBG_get0_parent.pod``` (point: new name) I can do that.

*HOWEVER* first the project has to decide it wants this.  As I've emailed, https://mta.openssl.org/pipermail/openssl-project/2020-June/002040.html, there's a thread started.

Reply:
On a technical note, the OPENSSL_CTX_get0_master_drbg() compat define is unnecessary because that function does not exist in 1.1.1.

Reply:
thanks.

Reply:
call them "Cric and Croc"

Reply:
Parent could be offensive to orphans.